packages/vuetify/src/composables/validation.ts

```diff
@@ -0,0 +1,80 @@
+// Utilities
+import { propsFactory, wrapInPromise } from '@/util'
+import { ref, unref, watch } from 'vue'
+
+// Types
+import type { MaybeRef } from '@/util/helpers'
+import type { PropType } from 'vue'
+
+export type ValidationRule = string | ((value: any) => string | boolean)
+export type ValidationRules = ValidationRule[]
+export type ValidationResult = string | boolean | Promise<string | boolean>
+
+export interface ValidationProps {
+  errorCount: string | number
+  rules: ValidationRules
+}
+
+export const makeValidationProps = propsFactory({
+  errorCount: {
+    type: [Number, String],
+    default: 1,
+  },
+  rules: {
+    type: Array as PropType<ValidationRules>,
+    default: () => ([]),
+  },
+})
+
+export function useValidation (props: ValidationProps, value: MaybeRef<any>) {
+  const errorMessages = ref<string[]>([])
+  const isPristine = ref(true)
+  const isValid = ref<null | boolean>(null)
+  const isValidating = ref(false)
+  const errorCount = ref(0)
+  const unwrefed = unref(value)
```

Comment:
I can't really make sense of why this is done, and why in function argument value is `MaybeRef`?

I think the input value should always be a `Ref`.

Reply:
Left there from testing